zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -445,11 +445,11 @@ public void runAll() {
         AuthenticationInfo authenticationInfo = new AuthenticationInfo();
         authenticationInfo.setUser(cronExecutingUser);
         p.setAuthenticationInfo(authenticationInfo);
-
-        p.setListener(jobListenerFactory.getParagraphJobListener(this));
-        Interpreter intp = factory.getInterpreter(getId(), p.getRequiredReplName());
-
-        intp.getScheduler().submit(p);
+        try {
+          run(p.getId());
+        } catch (InterpreterException intpException) {
+          throw intpException;
+        }
```

Comment:
What is the point of `try..catch` block here? 

Could you please explain, how is this code different from: 

```java
run(p.getId());
```